akka-persistence/src/test/scala/akka/persistence/EventAdapterSpec.scala

```diff
@@ -0,0 +1,204 @@
+      case e                                       ⇒ MyTagJournal.NotTagged(e)
+    }
+    override def fromJournal(event: MyTagJournal.JournalModel): Any =
+      event.payload
+  }
+
+  class ReplayPassThroughAdapter extends UserAgeTaggingAdapter {
+    override def fromJournal(event: MyTagJournal.JournalModel): Any =
+      event // don't unpack, just pass through the JournalModel
+  }
+
+  // TODO better split out EventReadAdapter and WriteAdapter ???
+  class PayloadUnwrapper extends EventAdapter[JournalModel] {
+    override def toJournal(event: Any): JournalModel = MyTagJournal.NotTagged(event)
+    override def fromJournal(event: JournalModel): Any = event.payload
+  }
+
+  class LoggingAdapter(system: ActorSystem) extends EventAdapter[Any] {
+    final val log = Logging(system, getClass)
+    override def toJournal(event: Any): Any = {
+      log.info("On its way to the journal: []: " + event)
+      event
+    }
+    override def fromJournal(event: Any): Any = {
+      log.info("On its way out from the journal: []: " + event)
+      event
+    }
+  }
+
+  case object GetState
+  class PersistAllIncomingActor(name: String, override val journalPluginId: String)
+    extends NamedPersistentActor(name) with PersistentActor {
+
+    var state: List[Any] = Nil
+
+    val persistIncoming: Receive = {
+      case GetState ⇒
+        state.reverse.foreach { sender() ! _ }
+      case in ⇒
+        persist(in) { e ⇒
+          state ::= e
+          sender() ! e
+        }
+    }
+
+    override def receiveRecover = {
+      case RecoveryCompleted ⇒ // ignore
+      case e                 ⇒ state ::= e
+    }
+    override def receiveCommand = persistIncoming
+  }
+
+}
+
+abstract class EventAdapterSpec(journalName: String, journalConfig: Config, adapterConfig: Config)
+  extends AkkaSpec(journalConfig.withFallback(adapterConfig))
+  with PersistenceSpec with ImplicitSender {
+  import EventAdapterSpec._
+
+  def this(journalName: String, journalConfig: Config) {
+    this(journalName, journalConfig, ConfigFactory.parseString(
+      s"""
+         |akka.persistence.journal {
+         |  event-adapters {
+         |    age                 = "${classOf[EventAdapterSpec].getCanonicalName}$$UserAgeTaggingAdapter"
+         |    logging             = "${classOf[EventAdapterSpec].getCanonicalName}$$LoggingAdapter"
+         |    replay-pass-through = "${classOf[EventAdapterSpec].getCanonicalName}$$ReplayPassThroughAdapter"
+         |    unwrap              = "${classOf[EventAdapterSpec].getCanonicalName}$$PayloadUnwrapper"
+         |  }
+         |
+         |  # define bindings for all journals
+         |  event-adapter-bindings {
+         |    "${classOf[EventAdapterSpec].getCanonicalName}$$UserDataChanged"   = age
+         |    "${classOf[EventAdapterSpec].getCanonicalName}$$TaggedDataChanged" = age
+         |    "${classOf[EventAdapterSpec].getCanonicalName}$$MyTagJournal$$JournalModel"      = unwrap
+         |  }
+         |
```

Comment:
@patriknw so that's the style of configuration we can do:

- adapters defined in one place
- bindings defined at top level, or:
- for each plugin separately